### PR TITLE
Reduced the use of ObjectManager

### PR DESCRIPTION
**Pull request by HansSchouten**
### Description
This pull request replaces a number of calls to the ObjectManager by constructor dependency injection.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Sales/Model/AdminOrder/Create.php [hunk 1]
```diff
@@ -232,6 +232,11 @@ class Create extends \Magento\Framework\DataObject implements \Magento\Checkout\
     private $serializer;
 
     /**
+     * @var \Magento\Framework\Locale\FormatInterface $localeFormat
+     */
+    protected $localeFormat;
```
Comment (maghamed) [inline app/code/Magento/Sales/Model/AdminOrder/Create.php]: all new properties added should be private.
Inheritance based APIs are highly discouraged

File: app/code/Magento/Theme/Model/View/Design.php [hunk 2]
```diff
@@ -74,6 +74,11 @@ class Design implements \Magento\Framework\View\DesignInterface
     protected $_appState;
 
     /**
+     * @var \Magento\Framework\Locale\ResolverInterface
+     */
+    protected $localeResolver;
```
Comment (maghamed) [inline app/code/Magento/Theme/Model/View/Design.php]: same as above, use private

**Review by maghamed — CHANGES_REQUESTED** [review]

**Comment by ishakhsuvarov** [comment]
@Hansschouten Please check the unit test failures on Travis CI.
Thanks!

**Comment by ishakhsuvarov** [comment]
@Hansschouten Closing this PR because of inactivity.
Please reopen and sync with the latest `develop` branch if you would like to continue.
Thanks!

**Pull request closed without merging**